### ISSUE TITLE
Systemd doesn't create alias symlinks according to its own rules

### ISSUE THREAD

**Issue by mobejm** [systemd/systemd#33513]
### systemd version the issue has been seen with

252

### Used distribution

systemd 252 (252-32.0.2.el9_4)

### Linux kernel version used

5.15.0-206.153.7.1.el9uek.x86_64

### CPU architectures issue was seen on

x86_64

### Component

systemctl

### Expected behaviour you didn't see

When enabling a unit located outside of systemd's search paths, if such unit has an Alias defined, the alias symlink is not created correctly, resulting in `systemctl` being unable to obtain the status of the service using such alias.

Given a unit file that looks something like this:

```
...
[Install]
Alias=myalias.service
...
```

The expectation is that executing the command `systemctl enable /etc/myservice/myservice.service` (or calling the corresponding dbus API - EnableUnitFiles) would result in two symlinks being created under `/etc/systemd/system`:

```
/etc/systemd/system/myservice.service -> /etc/myservice/myservice.service
/etc/systemd/system/myalias.service -> /etc/systemd/system/myservice.service
```

### Unexpected behaviour you saw

Given the unit file from above, the following two symlinks are created instead:

```
/etc/systemd/system/myservice.service -> /etc/myservice/myservice.service
/etc/systemd/system/myalias.service -> /etc/myservice/myservice.service
```

The fact that the alias symlink points to the original location of the unit file (outside of the search paths), makes systemd unable to recognize the status of the service using the alias.

```
$ systemctl status myalias
○ myalias.service - {"a":"...","v":"..."}
     Loaded: loaded (/etc/systemd/system/myalias.service; alias)
     Active: inactive (dead)
```

The output from above is unexpected given the service is enabled and active:

```
$ systemctl status myservice
○ myservice.service - {"a":"...","v":"..."}
     Loaded: loaded (/etc/systemd/system/myservice.service; enabled; preset: disabled)
...
```

This is unexpected given the official documentation of the `EnableUnitFiles` API:

> EnableUnitFiles() may be used to enable one or more units in the system (by creating symlinks to them in /etc or /run). It takes a list of unit files to enable (**either just file names or full absolute paths if the unit files are residing outside the usual unit search paths**), and two booleans: the first controls whether the unit shall be enabled for runtime only (true, /run), or persistently (false, /etc). The second one controls whether symlinks pointing to other units shall be replaced if necessary. This call returns one boolean and an array with the changes made. The boolean signals whether the unit files contained any enablement information (i.e. an [Install]) section. The changes list consists of structures with three strings: the type of the change (one of symlink or unlink), the file name of the symlink and the destination of the symlink. Note that most of the following calls return a changes list in the same format.

(Source: https://www.freedesktop.org/wiki/Software/systemd/dbus/)

The documentation for the `systemctl enable` command also seems to suggest that this behavior is not expected:

> enable UNIT…, enable PATH…
>
>[...]
>
>    This command expects either valid unit names (in which case various unit file directories are automatically searched for unit files with appropriate names), or absolute paths to unit files (in which case these files are read directly). If a specified unit file is located outside of the usual unit file directories, an additional symlink is created, linking it into the unit configuration path, thus ensuring it is found when requested by commands such as start. The file system where the linked unit files are located must be accessible when systemd is started (e.g. anything underneath /home/ or /var/ is not allowed, unless those directories are located on the root file system).
>
>[...]
>
>    Note that this operation creates only the symlinks suggested in the [Install] section of the unit files. While this command is the recommended way to manipulate the unit configuration directory, the administrator is free to make additional changes manually by placing or removing symlinks below this directory. This is particularly useful to create configurations that deviate from the suggested default installation. In this case, the administrator must make sure to invoke daemon-reload manually as necessary, in order to ensure the changes are taken into account.
>
>[...]

(Source: https://www.freedesktop.org/software/systemd/man/latest/systemctl.html)

### Steps to reproduce the problem

Simply create a unit file with an alias outside of systemd's search paths and try to enable it using its full path.

**Comment by poettering** [comment]
*[Canned reply follows]*

This is the upstream bug and feature request tracker of systemd. Please use this only for issues in the two most current upstream systemd versions. See this link for the list of current releases.

https://github.com/systemd/systemd/releases

For support for older versions please contact your distribution instead.

In order to keep frustration at a minimum the submission form tried to make this all very clear.

**Comment by joooeey** [comment]
This issue should be fixed in version 256 of systemd: #33565
